internal/cmd/commands/managedgroupscmd/ldap_funcs.go

```diff
@@ -0,0 +1,83 @@
+package managedgroupscmd
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/boundary/api/managedgroups"
+	"github.com/hashicorp/boundary/internal/cmd/base"
+)
+
+const (
+	groupNamesFlagName = "group-names"
+)
+
+type extraLdapCmdVars struct {
+	flagGroupNames []string
+}
+
+func init() {
+	extraLdapFlagsFunc = extraLdapFlagsFuncImpl
+	extraLdapActionsFlagsMapFunc = extraLdapActionsFlagsMapFuncImpl
+	extraLdapFlagsHandlingFunc = extraLdapFlagsHandlingFuncImpl
+}
+
+func extraLdapActionsFlagsMapFuncImpl() map[string][]string {
+	return map[string][]string{
+		"create": {groupNamesFlagName},
+		"update": {groupNamesFlagName},
+	}
+}
+
+func (c *LdapCommand) extraLdapHelpFunc(helpMap map[string]func() string) string {
+	var helpStr string
+	switch c.Func {
+	case "create":
+		helpStr = base.WrapForHelpText([]string{
+			"Usage: boundary managed-groups create ldap [options] [args]",
+			"",
+			"  Create a ldap-type managed group. Example:",
+			"",
+			`    $ boundary managed-groups create ldap -group-names admin -description "Ldap managed group for ProdOps"`,
+			"",
+			"",
+		})
+
+	case "update":
+		helpStr = base.WrapForHelpText([]string{
+			"Usage: boundary managed-groups update ldap [options] [args]",
+			"",
+			"  Update a ldap-type managed group given its ID. Example:",
+			"",
+			`    $ boundary managed-groups update ldap -id acctldap_1234567890 -name "devops" -description "Ldap managed group for DevOps"`,
+			"",
+			"",
+		})
+	}
+	return helpStr + c.Flags().Help()
+}
+
+func extraLdapFlagsFuncImpl(c *LdapCommand, _ *base.FlagSets, f *base.FlagSet) {
+	for _, name := range flagsLdapMap[c.Func] {
+		switch name {
+		case groupNamesFlagName:
+			f.StringSliceVar(&base.StringSliceVar{
+				Name:   groupNamesFlagName,
+				Target: &c.flagGroupNames,
+				Usage:  "The LDAP group names against which an LDAP account's associated groups (discovered during login) will be evaluated to determine membership (required). May be specified multiple times",
+			})
+		}
+	}
+}
+
+func extraLdapFlagsHandlingFuncImpl(c *LdapCommand, _ *base.FlagSets, opts *[]managedgroups.Option) bool {
+	switch {
+	case len(c.flagGroupNames) == 0:
+	case len(c.flagGroupNames) == 1 && c.flagGroupNames[0] == "null":
+		c.UI.Error(fmt.Sprintf("There must be at least one %q", groupNamesFlagName))
+		return false
```

Comment:
Two questions:
1. If the error is true, why is `len(c.flagGroupNames) == 0` a success case?
2. A bit ridicolous perhaps, but what's stopping someone from specifying 2 null valued groups? If we don't bother checking for that, why are we checking for the single case?

Reply:
I'm confused by the "error is true" portion of question #1.    If there's no group names, then it's a no-op.

Our CLI is consistent across several use-cases that in order to set a multi-value flag to `null` there for likely deleting all of current values for the flag, the user just needs to set the first value for the flag to `null`

Reply:
I see, that makes sense! Apologies for the confusion, my comment was regarding the `There must be at least one ...`, I was saying, "if that message is true, why is it ok to add 0?". Are you saying that it shouldn't be possible to unset group names once one has been set (which is the case the error is handling)?

Reply:
Ah... yes, we should return and error here.  Ty!

Reply:
Wait... you need to be able to update Name and Description and leave the existing group names as is.

Reply:
Sounds good to me